src/wordvec.c

```diff
@@ -0,0 +1,50 @@
+#define _GNU_SOURCE
+#include "wordvec.h"
+
+struct wordvec *wordvec_new(void)
+{
+	struct wordvec *wv = malloc(sizeof(*wv));
+	if (!wv)
+		return NULL;
+
+	wv->len = 0;
+	wv->cap = 8;
+
+	wv->vec = calloc(wv->cap, sizeof(*wv->vec));
+	if (!wv->vec) {
+		free(wv);
+		return NULL;
+	}
+
+	return wv;
+}
+
+void wordvec_del(struct wordvec *wv)
+{
+	free(wv->vec);
+	free(wv);
+}
+
+
+static struct wordvec *double_capacity(struct wordvec *wv)
+{
+	wv->cap *= 2;
+
+	wv->vec = reallocarray(wv->vec, wv->cap, sizeof(*wv->vec));
+	if (!wv->vec)
+		return NULL;
+	
+	return wv;
+}
+
+int wordvec_append(struct wordvec *wv, char c)
+{
+	if (wv->len >= wv->cap - 1)
```

Comment:
```suggestion
	if (wv->len > wv->cap - 1)
```

So we don't realloc an extra time?

Reply:
We need the equal, we can only append if `wv->len < wv->cap - 1` (minus 1 for nul).
Therefore we have to check for superiority and equality (technically only for equality for it's prettier with `>=`)